src/libaktualizr/config/config.cc

```diff
@@ -248,15 +253,19 @@ void Config::updateFromCommandLine(const boost::program_options::variables_map&
   if (cmd.count("primary-ecu-hardware-id") != 0) {
     provision.primary_ecu_hardware_id = cmd["primary-ecu-hardware-id"].as<std::string>();
   }
-  if (cmd.count("secondary-config") != 0) {
-    std::vector<boost::filesystem::path> sconfigs = cmd["secondary-config"].as<std::vector<boost::filesystem::path>>();
-    readSecondaryConfigs(sconfigs);
+  if (cmd.count("secondary-configs-dir") != 0) {
+    uptane.secondary_configs_dir = cmd["secondary-configs-dir"].as<boost::filesystem::path>();
   }
 }
 
-void Config::readSecondaryConfigs(const std::vector<boost::filesystem::path>& sconfigs) {
-  for (auto it = sconfigs.cbegin(); it != sconfigs.cend(); ++it) {
-    uptane.secondary_configs.emplace_back(Uptane::SecondaryConfig(*it));
+void Config::readSecondaryConfigs(const boost::filesystem::path& sconfigs_dir) {
+  if (!boost::filesystem::is_directory(sconfigs_dir)) {
+    LOG_ERROR << "Could not read secondary configs from " << sconfigs_dir << ": not a directory";
+    return;
+  }
+  for (const auto& config_file : Utils::glob((sconfigs_dir / "*.json").string())) {
+    LOG_INFO << "Parsing secondary config: " << config_file;
+    uptane.secondary_configs.emplace_back(config_file);
```

Comment:
Maybe we should allow providing a single json file if the user wants to? As in, the target can be a directory or a single file.

Reply:
I think it would complicate the docs and the cli a little bit. Because we essentially use this feature for demos, I don't know if we need to be that flexible for the moment.

The primary goal was just to get rid of this ugly CLI-only feature and its associated hacks.

Reply:
That's fine. Not worth investing effort in something that no one needs. I'm happy with this upgrade as it is.